File: wr-app/src/api/models/erp/custody.ts
export interface CustodyStockOption {
    assetId: number; assetCode: string; serialNumber: string | null; modelName: string; locationName: string;
}
export interface CustodyEquipmentSetOption { equipmentSetId: number; code: string; name: string; componentCount: number }
export interface CustodyItem {
    id: number; assetId: number | null; balanceId: number | null; equipmentSetId: number | null;
    equipmentSetCode: string | null; equipmentSetName: string | null; componentRole: string | null;
    assetCode: string; serialNumber: string | null; modelName: string; locationName: string;
    quantity: string; issueMovementId: number; returnedAt: string | null;
    returnConditionTypeId: number | null; returnConditionName: string | null; returnBlocksAvailability: boolean | null;
    inspectionNotes: string | null; returnInspectionId: number | null; maintenanceWorkOrderId: number | null;
}
export interface CustodyReturn {
    id: number; returnedAt: string; returnedById: number | null; returnedByLogin: string | null; itemIds: number[];
}
export interface Custody {
    id: number; organizationId: number; organizationName: string; unitId: number | null; unitName: string | null;
    recipientId: number; recipientName: string; authorizerId: number; authorizerName: string; purpose: string;
    status: string; deliveredAt: string; dueAt: string | null; completedAt: string | null;
    issuedById: number | null; issuedByLogin: string | null; items: CustodyItem[]; returns: CustodyReturn[];
}
export interface CustodyIssueRequest {
    requestId: string; organizationId: number; unitId?: number; recipientId: number; authorizerId: number;
    purpose: string; dueAt?: string; assetIds: number[]; equipmentSetIds: number[];
}
export interface CustodyPage<T> { content: T[]; totalElements: number; page: number; size: number }

File: wr-api/src/main/java/com/weaponsregistration/custody/model/Custody.java
package com.weaponsregistration.custody.model;

import com.weaponsregistration.core.model.*;
import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "erp_custody")
public class Custody extends CoreEntity {
    @ManyToOne(optional = false) @JoinColumn(name = "organization_id", nullable = false)
    public Organization organization;
    @ManyToOne @JoinColumn(name = "unit_id")
    public OrganizationalUnit unit;
    @ManyToOne(optional = false) @JoinColumn(name = "recipient_id", nullable = false)
    public Person recipient;
    @ManyToOne(optional = false) @JoinColumn(name = "authorizer_id", nullable = false)
    public Person authorizer;
    @Column(nullable = false) public String organizationName;
    @Column public String unitName;
    @Column(nullable = false) public String recipientName;
    @Column(nullable = false) public String authorizerName;
    @Column(nullable = false, length = 255) public String purpose;
    @Column(nullable = false, length = 30) public String status = "ACTIVE";
    @Column(nullable = false) public LocalDateTime deliveredAt;
    @Column public LocalDateTime dueAt;
    @Column public LocalDateTime completedAt;
    @Column public Long issuedById;
    @Column public String issuedByLogin;
    @Column(nullable = false, unique = true, length = 36) public String requestId;
    @Column(nullable = false, length = 64) public String requestFingerprint;
}
